app/javascript/src/utils/__tests__/search.test.js

```diff
@@ -223,4 +225,27 @@ describe('Search utilities', () => {
       );
     });
   });
+
+  describe('fetchSearchAPIPromise', () => {
+    let responsePromise;
+    let dataHash;
+
+    beforeEach(() => {
+      fetch.resetMocks();
+      fetch.once({});
+      dataHash = { name: 'jav' };
+      responsePromise = fetchSearchAPIPromise('tags', dataHash);
+    });
+
+    test('should return a Promise', () => {
+      expect(responsePromise).toBeInstanceOf(Promise);
+    });
+
+    test('should return response formatted as JSON', () => {
+      responsePromise.then(response => {
+        expect(response).toBeInstanceOf(Object);
+        expect(response).toMatchObject({ results: expect.any(Array) });
```

Comment:
shouldn't we be testing that `results` contains a specific result, rather than an instance of any `Array`? or am i misunderstanding what's happening here?

Reply:
I'm down for that, any suggestions on how to go about that? Since I'm mocking the request and this is in Jest, I'm not sure how to go about creating the necessary records and then making a real request to test for that.

My thinking was the Ruby specs cover this case already by checking the returned results.

Reply:
Ah I wrote that with the assumption that we already had mocks for this stuff 😂

Reply:
I'd be curious to hear if @nickytonline has plans for or thoughts around this.

But I don't think we need to block on this PR just for this test.

Reply:
I really like that idea! I was initially going to request his review, but he's out on vacation still I believe.

Reply:
I'll approve but I think we can ping him when he's back (or he might see my mention and jump right in 😉)

Reply:
@vaidehijoshi @atsmith813, I was planning on making a generic fetch utility with common stuff we always pass in, e.g. headers and maybe some other options. Merge this as is, and I will work on the generic fetch util when I'm officially back Wednesday. 👋